website/addons/dryad/static/dryadBrowser.js

```diff
@@ -0,0 +1,199 @@
+'use strict';
+
+var ko = require('knockout');
+var bootbox = require('bootbox');
+require('knockout.punches');
+var Raven = require('raven-js');
+var $osf = require('js/osfHelpers');
+
+function ViewModel(url, nodeSettings) {
+    var self = this;
+    self.nodeSettings = nodeSettings;
+    self.url = url;
+    self.urls = ko.observable();
+    self.inSearchMode = false;
+    self.doi = ko.observable("");
+    self.totalResults = ko.observable(0);
+    self.firstIndex = ko.observable(0);
+    self.lastIndex = ko.observable(0);
+    self.packages = ko.observableArray();
+    self.searchTerms  = ko.observable("");
+
+    self.message = ko.observable("");
+    self.messageClass = ko.observable("");
+
+    self.messages = {
+        browseError: ko.pureComputed(function() {
+            return 'Could not retrieve settings. Please refresh the page or ' +
+                'contact <a href="mailto: support@osf.io">support@osf.io</a> if the ' +
+                'problem persists.';
+        }),
+        searchError: ko.pureComputed(function(){
+            return 'DOI Successfully Validated: '+self.doi();
+        }),
+        noResultsError: ko.pureComputed(function(){
+            return 'Could Not Validate DOI: '+self.doi();
+        }),
+        communicationFailure: ko.pureComputed(function(){
+            return "Failed to Communicate with Dryad Servers. Try Again later.";
+        }),
+    };
+}
+
+ViewModel.prototype.init = function(){
+    var self=this;
+    $.getJSON(self.url).done(function(response){
+        self.urls(response.urls);
+        self.browseTo(20,0);
+    }).fail(function(xhr, textStatus, error){
+        self.changeMessage(self.messages.browseError(),
+        'text-danger');
+        Raven.captureMessage(self.messages.browseError(), {
+            extra: {
+                url: self.url,
+                textStatus: textStatus,
+                error: error
+                }
+        });
+    });
+}
+
+ViewModel.prototype.changeMessage = function(text, css, timeout) {
```

Comment:
The `changeMessageMixin` is the preferred way of adding this functionality to viewmodels. See `s3UserConfig.js` or `accountSettings.js`